dvc/scm/git/__init__.py

```diff
@@ -244,34 +247,42 @@ def belongs_to_scm(self, path):
         return basename == self.ignore_file or Git.GIT_DIR in path_parts
 
     def get_tree(self, rev):
-        return GitTree(self.git, rev)
+        return GitTree(self.repo, rev)
 
     def _get_diff_trees(self, a_ref, b_ref):
+        """Private method for getting the trees and commit hashes of 2 git
+        references.
+        Requires `gitdb` module (from gitpython package).
+
+        Args:
+            a_ref(str) - git reference
+            b_ref(str) - second git reference. If None, uses HEAD
 
+        Returns:
+            tuple - tuple with elements: (trees, commits)
+        """
         from gitdb.exc import BadObject, BadName
 
         trees = {DIFF_A_TREE: None, DIFF_B_TREE: None}
         commits = []
+        if b_ref is None:
+            b_ref = self.repo.head.commit
         try:
-            if b_ref is not None:
-                a_commit = self.git.commit(a_ref)
-                b_commit = self.git.commit(b_ref)
-                commits.append(str(a_commit))
-                commits.append(str(b_commit))
-            else:
-                a_commit = self.git.commit(a_ref)
-                b_commit = self.git.head.commit
-                commits.append(str(a_commit))
-                commits.append(str(b_commit))
+            a_commit = self.repo.commit(a_ref)
+            b_commit = self.repo.commit(b_ref)
+            # See https://gitpython.readthedocs.io
+            # /en/2.1.11/reference.html#git.objects.base.Object.__str__
+            commits.append(str(a_commit)[:7])
+            commits.append(str(b_commit)[:7])
```

Comment:
...which I've moved to here, including the hardcoded 7-len string conversion.

This also affects several test files which I had to change to also have hardcoded 7-len SHA strings or they will fail... Not sure this is acceptable though, please advise guys.